PSI Food waste/Startup.cs

```diff
@@ -32,6 +33,9 @@ public void ConfigureServices(IServiceCollection services)
             services.AddSession();
             services.AddSingleton<IProductRepository, ProductService>();
             services.AddSingleton<IRestaurantRepository, RestaurantServices>();
+            services.AddSingleton<IRegistrationEventNotifier, RegistrationEventNotifier>();
+            services.AddTransient<INotificationEvent, NotificationEvent>();
```

Comment:
Is it necessary for it to be transient? I think Singleton would work here as well

Reply:
Not really, because transient services are disposed at the end of request, while singleton is used throughout the whole lifespam of the app, so singleton takes much more memory, i think